api/util_test.go

```diff
@@ -0,0 +1,47 @@
+package api
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCallWithTimeoutTimesOut(t *testing.T) {
+	done := make(chan struct{})
+	defer func() {
+		close(done)
+	}()
+
+	logger, hook := test.NewNullLogger()
+	err := callWithTimeout(context.Background(), logger, 1*time.Nanosecond, func(ctx context.Context) error {
+		<-done
+		return errors.New("should not return")
+	})
+
+	require.Error(t, err)
```

Comment:
this shouldn't be necessary also

Reply:
I'm OK with this one, it ensures that there's no panic if a subsequent check assumes the error is not nil. Maybe `ErrorIs` handles the nil case, or maybe not.